src/Entity/Mooc/BaseMoocElement.php

```diff
@@ -116,4 +168,57 @@ public function detachChapter(): void
     {
         $this->chapter = null;
     }
+
+    /**
+     * @return Collection|AttachmentLink[]
+     */
+    public function getLinks(): Collection
+    {
+        return $this->links;
+    }
+
+    public function addLink(AttachmentLink $link): void
+    {
+        if (!$this->links->contains($link)) {
+            $this->links->add($link);
+        }
+    }
+
+    public function removeLink(AttachmentLink $link): void
+    {
+        $this->links->removeElement($link);
+    }
+
+    /**
+     * @return Collection|AttachmentFile[]
+     */
+    public function getFiles(): Collection
+    {
+        return $this->files;
+    }
+
+    public function addFile(AttachmentFile $file): void
+    {
+        if (!$this->files->contains($file)) {
+            $this->files->add($file);
+            $file->setType($this->getType());
```

Comment:
why do you need to set the type of file here?

Reply:
to create the path depending of the type of this object , @thomasbisignani  need it